diff --git a/pages/auth/login.vue b/pages/auth/login.vue
index 5c4ea98..aece15e 100644
--- a/pages/auth/login.vue
+++ b/pages/auth/login.vue
@@ -11,8 +11,8 @@ const route = useRoute()
 
 const method = ref('accessKeyAndSecretKey')
 const accessKeyAndSecretKey = ref({
-  accessKeyId: '',
-  secretAccessKey: '',
+  accessKeyId: 'rustfsadmin_name',
+  secretAccessKey: 'rustfsadmin_pswd',
 })
 
 const sts = ref({
