diff --git a/rustfs/src/main.rs b/rustfs/src/main.rs
index 1a95efe..c8e4477 100644
--- a/rustfs/src/main.rs
+++ b/rustfs/src/main.rs
@@ -190,7 +190,7 @@ async fn run(opt: config::Opt) -> Result<()> {
     let heal_manager = init_heal_manager(heal_storage, None).await?;
 
     let scanner = Scanner::new(Some(ScannerConfig::default()), Some(heal_manager));
-    scanner.start().await?;
+    // scanner.start().await?;
     print_server_info();
     init_bucket_replication_pool().await;
 
