diff --git a/pages/browser/[bucket]/[[key]].vue b/pages/browser/[bucket]/[[key]].vue
index 5b77db4..a06e5c0 100644
--- a/pages/browser/[bucket]/[[key]].vue
+++ b/pages/browser/[bucket]/[[key]].vue
@@ -21,7 +21,7 @@
 <script lang="ts" setup>
 
 import { useRoute } from '#app'
-import { endsWith } from 'lodash'
+import { endsWith } from 'lodash-es'
 import { joinRelativeURL } from 'ufo'
 import { computed } from 'vue'
