diff --git a/rustfs/Cargo.toml b/rustfs/Cargo.toml
index 018b279..9ff8d2c 100644
--- a/rustfs/Cargo.toml
+++ b/rustfs/Cargo.toml
@@ -131,7 +131,6 @@ sysctl = { workspace = true }
 [target.'cfg(target_os = "linux")'.dependencies]
 libsystemd.workspace = true
 
-[target.'cfg(all(target_os = "linux", target_env = "musl"))'.dependencies]
 mimalloc = { workspace = true }
 [target.'cfg(all(target_os = "linux", target_env = "gnu"))'.dependencies]
 tikv-jemallocator = { workspace = true }
diff --git a/rustfs/src/main.rs b/rustfs/src/main.rs
index f5ae2c5..53e24be 100644
--- a/rustfs/src/main.rs
+++ b/rustfs/src/main.rs
@@ -70,11 +70,6 @@ use std::sync::Arc;
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, error, info, instrument, warn};
 
-#[cfg(all(target_os = "linux", target_env = "gnu"))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
-
-#[cfg(all(target_os = "linux", target_env = "musl"))]
 #[global_allocator]
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
