diff --git a/rustfs/Cargo.toml b/rustfs/Cargo.toml
index 761ee3c..515f806 100644
--- a/rustfs/Cargo.toml
+++ b/rustfs/Cargo.toml
@@ -119,12 +120,6 @@ sysctl = { workspace = true }
 [target.'cfg(target_os = "linux")'.dependencies]
 libsystemd.workspace = true
 
-[target.'cfg(all(target_os = "linux", target_env = "gnu"))'.dependencies]
-tikv-jemallocator = "0.6"
-
-[target.'cfg(all(target_os = "linux", target_env = "musl"))'.dependencies]
-mimalloc = "0.1"
-
 [target.'cfg(not(target_os = "windows"))'.dependencies]
 pprof = { version = "0.15.0", features = ["flamegraph", "protobuf-codec"] }
 
diff --git a/rustfs/src/main.rs b/rustfs/src/main.rs
index 056e0dd..e22a59e 100644
--- a/rustfs/src/main.rs
+++ b/rustfs/src/main.rs
@@ -72,11 +72,6 @@ use std::str::FromStr;
 use std::sync::Arc;
 use tracing::{debug, error, info, instrument, warn};
 
-#[cfg(all(target_os = "linux", target_env = "gnu"))]
-#[global_allocator]
-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
-
-#[cfg(all(target_os = "linux", target_env = "musl"))]
 #[global_allocator]
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
